extension/goplugin/environment.go

```diff
@@ -356,17 +356,22 @@ func dispatchSchemaEventForEnv(env IEnvironment, prioritizedSchemaHandlers Prior
 		}
 	}
 	for _, priority := range sortSchemaHandlers(prioritizedSchemaHandlers) {
-		for _, schemaEventHandler := range prioritizedSchemaHandlers[priority] {
+		for index, schemaEventHandler := range prioritizedSchemaHandlers[priority] {
 			context["go_validation"] = true
 			if err := schemaEventHandler(context, resource, env); err != nil {
-				return err
+				env.Logger().Warningf("failed to handle schema '%s' event '%s' at priority '%d' with index '%d': %s", sch.ID(), event, priority, index, err)
+				switch err.(type) {
+				case goext.Error:
+					return err
+				default:
+					return goext.NewError(goext.ErrorBadRequest, err)
```

Comment:
I think all unspecified error should be handled as `InternalServerError`

Reply:
After offline discussion we came to a conclusion that the best solution is to enforce schema handlers to return always goext.Error and always define proper error. I changed the code accordingly. What do you think about this idea?

Reply:
Generally, it's good idea. Actually, "notification" event does not require status_code so please make sure this change works with cron or sync_watch case.